### PR TITLE
Add query/ref status to graphml file

### PR DESCRIPTION
**Pull request by muppi1993**
This will provide the information the frontend needs to highlight query samples in the network graph.

### CODE REVIEW

File: beebop/utils.py [hunk 1]
```diff
@@ -94,3 +94,29 @@ def replace_filehashes(folder, filename_dict):
                 if f_key in line:
                     line = line.replace(f_key, f_value)
             print(line)
+
+
+def add_query_ref_status(fs, p_hash, filename_dict):
+    ET.register_namespace('', "http://graphml.graphdrawing.org/xmlns")
+    ET.register_namespace('xsi', "http://www.w3.org/2001/XMLSchema-instance")
```
Comment (richfitz) [inline beebop/utils.py:101]: should these registrations be done at the top-level? I am not familiar with the package at all

File: beebop/utils.py [hunk 2]
```diff
@@ -94,3 +94,29 @@ def replace_filehashes(folder, filename_dict):
...
+    # list of query filenames
+    query_names = list(filename_dict.values())
+    # list of all component graph filenames
+    file_list = []
+    for file in os.listdir(fs.output_network(p_hash)):
+        if fnmatch.fnmatch(file, 'network_component_*.graphml'):
+            file_list.append(file)
```
Comment (richfitz) [inline beebop/utils.py:108]:
```suggestion
    file_list = glob.glob("network_component_*.graphml")
```

With the glob module this should work (https://docs.python.org/3/library/glob.html#glob.glob)

File: beebop/utils.py [hunk 3]
```diff
@@ -94,3 +94,29 @@ def replace_filehashes(folder, filename_dict):
...
+        nodes = graph.findall(".//{http://graphml.graphdrawing.org/xmlns}node")
+        for node in nodes:
+            name = node.find("./").text
+            child = ET.Element("data")
+            child.set("key", "ref_query")
+            child.text = 'query' if name in query_names else 'ref'
+            node.append(child)
+        ET.indent(xml, space='  ', level=0)
+        with open(path, 'wb') as f:
+            xml.write(f, encoding='utf-8')
```
Comment (richfitz) [inline beebop/utils.py:122]: I see in the docs this is required but I always thought xml was defined to be UTF-8, odd that it defaults to us-ascii of all things

**Review by richfitz — CHANGES_REQUESTED** [review]
The code looks fine (a couple of small suggestions below) but it would be great to add some docstrings so someone coming later (e.g., me!) has a better idea of what this is trying to fix up in the data at a high level

**Comment by muppi1993** [comment]
I have added a docstring to explain what the function does. I think in view of the fact that I'm leaving soon, I'll add a separate ticket to improve the documentation overall, I've noticed that some docstrings are out of date, and other functions don't have any yet.

**Pull request merged**